node/tests/st/bgp/test_ipip.py

```diff
@@ -1,626 +0,0 @@
-
-        host1.start_calico_node("--backend={0}".format(backend))
-        host2.start_calico_node("--backend={0}".format(backend))
-
-        # Before creating any workloads, set the initial IP-in-IP state.
-        host1.set_ipip_enabled(with_ipip)
-
-        if rr_ip:
-            # Set the default AS number - as this is used by the RR mesh,
-            # and turn off the node-to-node mesh (do this from any host).
-            update_bgp_config(host1, asNum=64513, nodeMesh=False)
-            # Peer from each host to the route reflector.
-            for host in [host1, host2]:
-                create_bgp_peer(host, "node", rr_ip, 64513, metadata={'name':host.name})
-
-        # Create a network and a workload on each host.
-        network1 = host1.create_network("subnet1")
-        workload_host1 = host1.create_workload("workload1",
-                                               network=network1)
-        workload_host2 = host2.create_workload("workload2",
-                                               network=network1)
-
-        for _ in [1, 2]:
-            # Check we do or don't have connectivity between the workloads,
-            # according to the IP-in-IP setting.
-            if with_ipip:
-                # Allow network to converge.
-                self.assert_true(
-                    workload_host1.check_can_ping(workload_host2.ip, retries=10))
-
-                # Check connectivity in both directions
-                self.assert_ip_connectivity(workload_list=[workload_host1,
-                                                           workload_host2],
-                                            ip_pass_list=[workload_host1.ip,
-                                                          workload_host2.ip])
-
-                for host in [host1, host2]:
-
-                    # Check that we are using IP-in-IP for some routes.
-                    assert "tunl0" in host.execute("ip r")
-
-                    # Get current links and addresses, as a baseline for the
-                    # following changes test.
-                    host.execute("ip l")
-                    host.execute("ip a")
-
-                    if rr_ip:
-                        # Check that this host has received the /26 route from
-                        # the route reflector.  Otherwise this could show up
-                        # during the following monitoring, as it might be a
-                        # little slower than what is needed for the workload
-                        # connectivity tested above.
-                        expected_route_string = "/26 via " + rr_ip
-                        def check():
-                            self.assertIn(expected_route_string,
-                                          host.execute("ip r"))
-                        retry_until_success(check, retries=10)
-
-                    # Check that routes are not flapping, on either host, by
```

Comment:
This is something I remember as having been important once - i.e. we had a bug where the symptom was a persistent flapping with tunnel and non-tunnel routes.

Reply:
Yeah, I think a more detailed ip monitor test would be good to have. I've added a note to the doc to re-implement this, likely in the k8st suite.